src/slurm.jl

```diff
@@ -50,44 +50,46 @@ function launch(manager::SlurmManager, params::Dict, instances_arr::Array,
             mkdir(job_file_loc)
         end
 
-        println("removing old files")
+        # println("removing old files")
         # cleanup old files
-	map(f->rm(joinpath(job_file_loc, f)), filter(t -> occursin(r"job(.*?).out", t), readdir(job_file_loc)))
-        println("removing old Setting up srun commands")
+	    # map(f->rm(joinpath(job_file_loc, f)), filter(t -> occursin(r"job(.*?).out", t), readdir(job_file_loc)))
 
         np = manager.np
         jobname = "julia-$(getpid())"
-		jobID = String(ENV["SLURM_JOB_ID"])
-        srun_cmd = `srun -J $jobname -n $np -o "$(joinpath(job_file_loc, "job-$jobID-%4t.out"))" -D $exehome $(srunargs) $exename $exeflags $(worker_arg())`
+        job_output_name = "$(jobname)-$(trunc(Int, Base.time() * 10))"
+        make_job_output_path(task_num) = joinpath(job_file_loc, "$(job_output_name)-$(task_num).out")
+        job_output_template = make_job_output_path("%4t")
+        srun_cmd = `srun -J $jobname -n $np -o "$(job_output_template)" -D $exehome $(srunargs) $exename $exeflags $(worker_arg())`
         srun_proc = open(srun_cmd)
+        slurm_spec_regex = r"([\w]+):([\d]+)#(\d{1,3}.\d{1,3}.\d{1,3}.\d{1,3})"
         for i = 0:np - 1
             println("connecting to worker $(i + 1) out of $np")
-            local w=[]
-            fn = "$(joinpath(exehome, job_file_loc))/job-$jobID-$(lpad(i, 4, "0")).out"
+            local slurm_spec_match = nothing
+            fn = make_job_output_path(lpad(i, 4, "0"))
             t0 = time()
             while true
```

Comment:
I'm confused about the need for this loop. It looks like there's no situation where it would actually loop back to the top.

Reply:
The log file takes a while to create and might be empty for a split second when it is created. So you have to sit there checking to see if the file has been populated yet before you can parse it for the worker specification, and so that the code doesn't continue until the workers are actually created.

Reply:
Ahh, got it - can you just add a comment to that effect or something?